frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,491 @@
+              formValues={formValues}
+              handleChange={handleChange}
+            />
+            <RoutingLabelEditor
+              formValues={formValues}
+              setFormValues={setFormValues}
+              isDefaultReceiver={isDefaultReceiver}
+            />
+          </>
+        )}
+
+        <ButtonBar errorMessage={errorMsg} inProgress={inProgress}>
+          <ActionGroup className="pf-c-form">
+            <Button
+              type="submit"
+              variant="primary"
+              data-test-id="save-changes"
+              isDisabled={isFormInvalid as boolean}
+            >
+              {saveButtonText}
+            </Button>
+            <Button
+              type="button"
+              variant="secondary"
+              data-test-id="cancel"
+              onClick={history.goBack}
+            >
+              Cancel
+            </Button>
+          </ActionGroup>
+        </ButtonBar>
+      </form>
+    </div>
+  );
+};
+
+export const SaveAsDefaultCheckbox = ({ formField, disabled, label, formValues, handleChange }) => {
+  const saveAsDefaultLabelClass = classNames({ 'co-no-bold': disabled });
+  return (
+    <label className={saveAsDefaultLabelClass}>
+      <input
+        type="checkbox"
+        name={formField}
+        data-test-id="save-as-default"
+        onChange={(e) =>
+          handleChange({
+            target: { name: `${formField}`, value: e.target.checked },
+          })
+        }
+        checked={formValues[`${formField}`]}
+        disabled={disabled}
+      />
+      &nbsp; {label}
+    </label>
+  );
+};
+
+const ReceiverWrapper: React.FC<ReceiverFormsWrapperProps> = React.memo(({ obj, ...props }) => {
+  const { alertManagerBaseURL } = window.SERVER_FLAGS;
+  const [alertManagerGlobals, setAlertManagerGlobals] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMsg, setErrorMsg] = React.useState('');
+
+  if (_.isEmpty(errorMsg)) {
+    if (!alertManagerBaseURL) {
+      setErrorMsg(`Error alertManagerBaseURL not set`);
+      return;
+    }
+    if (!inProgress && _.isEmpty(alertManagerGlobals)) {
+      setInProgress(true);
+      coFetchJSON(`${alertManagerBaseURL}/api/v2/status/`).then((data) => {
+        const origAlertMngrConfigStr = _.get(data, 'config.original', {});
+        if (_.isEmpty(origAlertMngrConfigStr)) {
+          setErrorMsg('alertmanager.v2.status.config.original not found.');
+          setInProgress(false);
+          return;
+        }
+        let origAlertMngrConfig;
+        try {
+          origAlertMngrConfig = safeLoad(origAlertMngrConfigStr);
+        } catch (e) {
+          setErrorMsg(`could not convert alertmanager config.original yaml: ${e}`);
+          setInProgress(false);
+          return;
+        }
+        setAlertManagerGlobals(origAlertMngrConfig.global);
+        setErrorMsg('');
+        setInProgress(false);
+      });
+      return;
+    }
+  }
+
+  return inProgress ? (
```

Comment:
Using early returns like `if (inProgress) { ... }` or `if (errorMsg) { ... }` (or both) better matches the coding style we have elsewhere in Console.

Reply:
oh, ok.  When I was working on chargeback reports, metering, those code reviews advocated the conditional operator (condition ? true : false)

Reply:
Well, I think one `? :` is fine as long as the blocks are not too big. In this case there's 3 blocks, so if you make one an early return, I think that is fine.

Reply:
fixed